runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java

```diff
@@ -171,6 +193,83 @@ public static Environment createProcessEnvironment(
     }
   }
 
+  public static Collection<ArtifactInformation> getArtifacts(PipelineOptions options) {
+    Set<String> pathsToStage = Sets.newHashSet();
+    List<String> experiments = options.as(ExperimentalOptions.class).getExperiments();
+    if (experiments != null) {
```

Comment:
Please add a TODO to remove this experiment when full solution is fully implemented (for all supported runners).

Reply:
done.

Reply:
done.